projects/ngx-sub-form/src/lib/ngx-sub-form-utils.ts

```diff
@@ -32,6 +34,37 @@ export type FormErrors<FormInterface> = null | Partial<
   }
 >;
 
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>; // can be removed when upgrading ts version
+type UntypedControlBase<TControl extends AbstractControl | FormArray> = Omit<
+  TControl,
+  'value' | 'valueChanges' | 'setValue' | 'patchValue'
+>;
+
+export type TypedAbstractControl<TValue extends any> = {
+  value: TValue;
+  valueChanges: Observable<TValue>;
+  setValue(
+    value: TValue,
+    options?: {
+      onlySelf?: boolean;
+      emitEvent?: boolean;
+      emitModelToViewChange?: boolean;
+      emitViewToModelChange?: boolean;
+    },
```

Comment:
Looking into Angular code base, `setValue` for an array only have the following props: `onlySelf`, `emitEvent`:

```
setValue(value: any[], options?: {
    onlySelf?: boolean;
    emitEvent?: boolean;
}): void;
```

If we want to make sure we're using the same values as Angular, one way of doing it could be:

```
Parameters<FormControl['setValue']>[1]
```

It feels a bit like magic so I do hesitate between that and just rewritting it to be honnest.

Here's a demo: https://www.typescriptlang.org/play/?ssl=10&ssc=66&pln=1&pc=1#code/KYDwDg9gTgLgBAE2AYwDYEMrDm9BnPOAMWgFsBhCAOxiglTgG8BYAKDjj2BgDV1UArsAAUAN35CAXHHRUAngBo4EMDACW1PAH5pLdh2VVUcgMrBUAMx1wARhHrBZAbjYG4wUmpgBRUcBrWdg7OrgYeXgCyEEioACoQPGrAAO7kABayAObAgfaojlQu+hzhvEnJ8VEx6Vk50kH5IfoAvgCU0qIQaghszWxsMHJg2CRQFNS09ADyqhpUhAC8cAAKmOik3MBQeAA8o+M0dKgA2gDkXLwSwKcAugB8xwCMN05AA

:point_up: over the `FormControlOptions` and see the definition :)

Reply:
Wups, my bad this was a C&P issue.

In regards to the `Parameters<FormControl['setValue']>[1]`:
- It was written like that at first (did I mention I'm in love with typescript type navigation ;) )
- But when testing the resulting intelli-sense I realized that because all properties of the parameter are optional the displayed type is `{}` :(
- That said, while writing this I just realized it might be possible to make all props required first and then Partial again`Partial<Required<Parameters<FormControl['setValue']>[1]>>`, maybe this way we can trick TS is spitting out the proper typing ... but as you mentioned we are moving into harry potter territory here :)

Reply:
> But when testing the resulting intelli-sense I realized that because all properties of the parameter are optional the displayed type is {}

I think it was working just fine for me :thinking: if you come up with a minimal repro I'm keen to look into it :)

Reply:
My mistake I used Parameters on `AbstractControl.setValue` where the signature of options is `Object`. Thats great newas as this means no need to C&P :)